src/user/handler.rs

```diff
@@ -1,69 +1,66 @@
-use crate::database::pool::PgPool;
+use crate::database::Pool;
 use crate::errors::ServiceError;
-use crate::user::manager::{user_manager_login, user_manager_register};
-use crate::user::model::{AuthData, LoggedUser, SlimUser, UserData};
-use actix_identity::Identity;
+use crate::user::model::{LoggedUser, SlimUser, UserData};
+use crate::user::service as user;
+use actix_identity::{Identity, RequestIdentity};
 use actix_web::dev::Payload;
-use actix_web::error::BlockingError;
 use actix_web::{web, Error, FromRequest, HttpRequest, HttpResponse};
-use futures::Future;
 
 impl FromRequest for LoggedUser {
     type Error = Error;
-    type Future = Result<LoggedUser, Error>;
+    type Future = futures::future::Ready<Result<Self, Self::Error>>;
     type Config = ();
 
-    fn from_request(req: &HttpRequest, pl: &mut Payload) -> Self::Future {
-        if let Some(identity) = Identity::from_request(req, pl)?.identity() {
-            let user: LoggedUser = serde_json::from_str(&identity)?;
-            return Ok(user);
-        }
+    fn from_request(req: &HttpRequest, _: &mut Payload) -> Self::Future {
+        let identity = req.get_identity();
 
-        Ok(LoggedUser { email: None })
+        let slim_user = if let Some(identity) = identity {
```

Comment:
I feel it strange to use `None` in `email` of a `LoggedUser` as a flag to say the User is not logged in...
I think we should directly use an `Option` wherever we use a `LoggedUser` => which I called `SlimUser`.

Reply:
Some bad habits coming from PHP and NULL, I should be careful about that!